### PR TITLE
Always set Parquet-Avro projection

### PR DESCRIPTION
**Pull request by clairemcginty**

### CODE REVIEW

**Review by RustedBones — COMMENTED** [review]
Can we test the opposite case, when we have an avro schema with default field missing from the parquet data ?

I fear in this case the projection will fail because the optional field is missing in parquet. Projection may have to be the perfect intersection of schema

**Comment by clairemcginty** [comment]
> Can we test the opposite case, when we have an avro schema with default field missing from the parquet data ?
> 
> I fear in this case the projection will fail because the optional field is missing in parquet. Projection may have to be the perfect intersection of schema

Added a new default field to the projection schema -- test still passes: https://github.com/spotify/scio/pull/5234/commits/f88d458a56b26b7f322a9889d0cc941e876695b0

File: scio-smb/src/test/avro/schema.avsc [hunk 1]
```diff
@@ -21,5 +21,22 @@
...
+    "name": "AvroGeneratedUserProjection",
+    "namespace": "com.spotify.scio.smb",
+    "doc": "Record for testing trimmed projections",
+    "fields": [
+        {
+          "name": "name",
+          "type": "string"
+        },
+        {
+          "name": "optionalFieldWithDefault", /* Not present in original AvroGeneratedUser schema */
```
Comment (RustedBones) [inline scio-smb/src/test/avro/schema.avsc:36]: I don't think comments are allowed. you should use prob use `doc` field

**Comment by RustedBones** [comment]
Can `tensorflow` package follow the same logic for consistency ?

**Comment by clairemcginty** [comment]
> Can `tensorflow` package follow the same logic for consistency ?

I think tensorflow package is a bit different, because there isn't a readSchema to fall back on AFAIK... https://github.com/spotify/scio/blob/v0.14.0/scio-parquet/src/main/scala/com/spotify/scio/parquet/tensorflow/ParquetExampleIO.scala#L76-L79

I can look into it a bit more/add a unit test, because I don't understand 100% how this works

**Review by RustedBones — APPROVED** [review]

**Pull request merged**